app/config/galoy-instances.ts

```diff
@@ -1,42 +1,84 @@
 import { NativeModules } from "react-native"
 
+// this is used for local development
+// will typically return localhost
 const scriptHostname = (): string => {
   const { scriptURL } = NativeModules.SourceCode
   const scriptHostname = scriptURL?.split("://")[1].split(":")[0] ?? ""
   return scriptHostname
 }
 
-export const possibleGaloyInstanceNames = ["BBW", "Staging", "Local", "Custom"] as const
+export const standardInstances = ["Main", "Staging", "Local"] as const
+export const possibleGaloyInstanceNames = [...standardInstances, "Custom"] as const
 export type GaloyInstanceName = (typeof possibleGaloyInstanceNames)[number]
 
+export type StandardInstance = {
+  id: "Main" | "Staging" | "Local"
+}
+
+export type CustomInstance = {
+  id: "Custom"
+  name: string
+  graphqlUri: string
+  graphqlWsUri: string
+  posUrl: string
+  lnAddressHostname: string
+  blockExplorer: string
+}
+
+export type GaloyInstanceInput = StandardInstance | CustomInstance
+
 export type GaloyInstance = {
-  name: GaloyInstanceName
+  id: GaloyInstanceName
+  name: string
   graphqlUri: string
   graphqlWsUri: string
   posUrl: string
   lnAddressHostname: string
+  blockExplorer: string
+}
+
+export const maybeAddDefault = (input: GaloyInstanceInput): GaloyInstance => {
+  if (input.id === "Custom") {
+    return input
+  }
+
+  const instance = GALOY_INSTANCES.find((instance) => instance.name === input.id)
+
+  if (instance) {
+    return instance
+  }
+
+  console.error("instance not found") // should not happen
+  return GALOY_INSTANCES[0]
 }
 
 export const GALOY_INSTANCES: GaloyInstance[] = [
```

Comment:
We could convert this to an object with the keys being the "Id" to avoid some of the typing issues.

Reply:
I gave it a try, ie:

```
const GALOY_INSTANCES = {
  Main: {
    name: "BBW",
    graphqlUri: "https://api.mainnet.galoy.io/graphql",
    graphqlWsUri: "wss://api.mainnet.galoy.io/graphql",
    posUrl: "https://pay.bbw.sv",
    lnAddressHostname: "pay.bbw.sv",
    blockExplorer: "https://mempool.space/tx/",
  },
  Staging: {
    name: "Staging",
    graphqlUri: "https://api.staging.galoy.io/graphql",
    graphqlWsUri: "wss://api.staging.galoy.io/graphql",
    posUrl: "https://pay.staging.galoy.io",
    lnAddressHostname: "pay.staging.galoy.io",
    blockExplorer: "https://mempool.space/signet/tx/",
  },
  Local: {
    name: "Staging",
    graphqlUri: `http://${scriptHostname()}:4002/graphql`,
    graphqlWsUri: `ws://${scriptHostname()}:4002/graphql`,
    posUrl: `http://${scriptHostname()}:3000`,
    lnAddressHostname: `${scriptHostname()}:3000`,
    blockExplorer: "https://mempool.space/signet/tx/",
  },
} as const
```

but because we have the Custom flow, at the end of the day it's not changing much the need for complex type. I'm going to stick with the Array for now but see if I can improve the type

Reply:
so the type issue is related to `find()` that always return `T | undefined`. there would be other ways to go with that does not rely on `find()` but that would make the code more complex (using filter and map). I think the current tradeoff is a good one.